doc/developer_guide/developer_guide.md

```diff
@@ -0,0 +1,49 @@
+# Script-Languages-Container-Tool Developer Guide
+
+r-exasol is a R library providing access to an Exasol DB. 
+This document is about the inner working of r-exasol.
+
+
+
+## Problem Statement:
+As described in the [README.md](../../README.md):
+>'It is optimised for fast reading & writing from and to a multinode cluster.'
+> 
+This document will describe how the r-exasol library reads/write data from/to the database from high level perspective.
+The focus is hereby on the functions which leverage the fast data transfer, as the other functions make usage of standard DBI / RODBC functions.
+
+## Implementation
+
+### Connection
+
+When the client calls dbConnect(), the library connects to the Exasol DB via RODBC, this happens in ```EXADBI.R```:
+```r
+    EXANewConnection <- function(...)
+    ...
+    odbcDriverConnect(...)
+```
+
+### Data Exchange
+
+Once the connection is ready, the client can read data or manipulate data using basically two functions:
+
+```r
+    exa.readData()
+    exa.writeData()
+```
+
+Some of the standard SQL functions use internally one of those two methods, for example ```dbGetQuery()```,
+but that is not always the case.
+
+### Internal data flow when reading from database
+
+The following diagram shows the internal data flow when the client invokes ```exa.readData()```:
+
+![](./images/r-exasol.png)
```

Comment:
Usually we use UML (via plantuml). So here a Sequence diagram.
However, since it's only for developers, I  think it's fine in this case.

Reply:
Thanks for the hint, will use it next time.